winsup/cygwin/create_posix_thread.cc

```diff
@@ -100,7 +100,22 @@ pthread_wrapper (PVOID arg)
 	   : : [WRAPPER_ARG] "o" (wrapper_arg),
 	       [CYGTLS] "i" (__CYGTLS_PADSIZE__));
 #elif defined(__aarch64__)
-  // TODO
+  __asm__ __volatile__ (
```

Comment:
Can please you explain why is `__volatile__` needed in this case? There does not seem to be any synchronization involved. E.g., can you give some example where the instructions re-ordering could cause the problem?